app/controllers/sections/transportArranger/TransportArrangerNameController.scala

```diff
@@ -67,22 +67,19 @@ class TransportArrangerNameController @Inject()(
     }
 
   private def withTransportArrangerAnswer(f: TransportArranger => Future[Result])(implicit request: DataRequest[_]): Future[Result] = {
-    request.userAnswers.get(TransportArrangerPage) match {
-      case Some(transportArranger) if transportArranger == GoodsOwner | transportArranger == Other => f(transportArranger)
-      case Some(transportArranger) =>
+    withAnswer(
+      page = TransportArrangerPage,
+      // TODO: update redirectRoute to journey index page when built
+      redirectRoute = controllers.sections.transportArranger.routes.TransportArrangerController.onPageLoad(request.ern, request.lrn, NormalMode)
+    ) {
+      case transportArranger if transportArranger == GoodsOwner | transportArranger == Other => f(transportArranger)
```

Comment:
A minor improvement if you want to make it 
```suggestion
      case transportArranger@(GoodsOwner | Other) => f(transportArranger)
```

Reply:
(thanks Eric)